charmrepo/charmstore_test.go

```diff
@@ -36,28 +36,15 @@ type charmStoreSuite struct {
 var _ = gc.Suite(&charmStoreSuite{})
 
 func (s *charmStoreSuite) TestURL(c *gc.C) {
-	store := newStoreRepo(c, "https://1.2.3.4/charmstore").(*charmrepo.CharmStore)
-	c.Assert(store.URL(), gc.Equals, "https://1.2.3.4/charmstore")
-}
-
-func (s *charmStoreSuite) TestDefaultURL(c *gc.C) {
-	store := newStoreRepo(c, "").(*charmrepo.CharmStore)
-	c.Assert(store.URL(), gc.Equals, csclient.ServerURL)
-}
-
-func (s *charmStoreSuite) TestCacheDir(c *gc.C) {
-	cacheDir := c.MkDir()
-	repo, err := charmrepo.NewCharmStore(charmrepo.NewCharmStoreParams{
-		CacheDir: cacheDir,
+	repo := charmrepo.NewCharmStore(charmrepo.NewCharmStoreParams{
+		URL: "https://1.2.3.4/charmstore",
```

Comment:
This part looks bad :( even just having additional newRepo(URL string) function locally would make the tests easier to read.